### PR TITLE
Update BaseCommand.requires_system_checks for django >=3.2

### PR DESCRIPTION
**Pull request by abhiabhi94**
# I have made things!
Updated the type definition of the `required_system_checks` attribute on the `BaseCommand`.

- the documentation for this specific change in django can be seen at https://docs.djangoproject.com/en/3.2/howto/custom-management-commands/#django.core.management.BaseCommand.requires_system_checks

## Related issues
None

### CODE REVIEW

**Comment by abhiabhi94** [comment]
As far I could see(I am shortsighted), i didn't see a test for the modules defined inside the `managment` package. If acceptable, I could write a test for the above change or other too maybe.

Also, I don't think the contributing instructions as of now mention the installation of `GDAL` as requirement for the running of tests. The test below actually seems to require it
https://github.com/typeddjango/django-stubs/blob/8f97bf880d3022772581ea4cf8b5bf5297a27bad/tests/typecheck/models/test_inheritance.yml#L94-L115

I can maybe add the link to [django's geospatial installation instructions](https://docs.djangoproject.com/en/dev/ref/contrib/gis/install/geolibs/) here, if acceptable.

File: django-stubs/core/management/base.pyi [hunk 1]
```diff
@@ -38,7 +39,7 @@ class BaseCommand:
     help: str = ...
     output_transaction: bool = ...
     requires_migrations_checks: bool = ...
-    requires_system_checks: bool = ...
+    requires_system_checks: Union[bool, List[Tags], Tuple[Tags], str]  = ...
```
Comment (sobolevn) [inline django-stubs/core/management/base.pyi]: Maybe `Tuple[Tags, ...]`?

**Review by sobolevn — COMMENTED** [review]
Thanks! Test would be awesome!

File: django-stubs/core/management/base.pyi (hunk 1 above)
Comment (abhiabhi94) [reply django-stubs/core/management/base.pyi]: yes, thanks for the feedback.

**Review by sobolevn — APPROVED** [review]

**Pull request merged**